src/client/Components/Util/OAuth.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+
+export default class OAuth extends React.Component {
+  state = {
+    user: {},
+    disabled: ''
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.startAuth = this.startAuth.bind(this);
+    this.closeCard = this.closeCard.bind(this);
```

Comment:
I'll add a comment to explain it, but basically you need to bind the function because it's a class function in React. This explains it better than I can: https://www.freecodecamp.org/news/this-is-why-we-need-to-bind-event-handlers-in-class-components-in-react-f7ea1a6f93eb/